--- cu-progs.m4.org
+++ cu-progs.m4
@@ -94,6 +94,9 @@
 gl_ADD_PROG([optional_bin_progs], [wc])
 gl_ADD_PROG([optional_bin_progs], [whoami])
 gl_ADD_PROG([optional_bin_progs], [yes])
+gl_ADD_PROG([optional_bin_progs], [crccp])
+gl_ADD_PROG([optional_bin_progs], [crcsum])
+gl_ADD_PROG([optional_bin_progs], [crcmv])
 no_install_progs_default='arch coreutils hostname'
 # Given the name of a variable containing a space-separated
 # list of install-by-default programs and the actual list of
